analyses/snv-callers/util/tmb_functions.R

```diff
@@ -72,92 +74,74 @@ snv_ranges_filter <- function(maf_df, keep_ranges = NULL, bp_window = 0) {
     keep_ranges,
     maxgap = bp_window
   )
-  
+
   # Calculate of ratio of variants in this BED using the @from slot which
-  # indicates the indices of the ranges in `wxs_maf_ranges` that have overlaps
-  # with `wxs_bed_ranges`
+  # indicates the indices of the ranges in `maf_ranges` that have overlaps
+  # with `keep_ranges`
   ratio <- length(overlap@from) / nrow(maf_df)
-  
+
   # What fraction of mutations are in these bed regions?
   cat(
     "Ratio of variants in this BED:", ratio, "\n",
     "Ratio of variants being filtered out:", 1 - ratio, "\n"
   )
-  
+
   # Only keep those in the BED regions that overlap the `wxs_bed_granges`
   filt_maf_df <- maf_df[unique(overlap@from), ]
-  
+
   # Return this matrix with the WXS mutations filtered but WGS the same
   return(filt_maf_df)
 }
 
-calculate_tmb <- function(maf_df, bed_wgs, bed_wxs) {
-  # Calculate Tumor Mutational Burden for each sample given their WGS or WXS
-  # sizes in bp. Based on the BED files provided, tthis function filters out SNV 
-  # outside those ranges and uses the total size in bp of those BED ranges for th
-  # TMB denominator. 
+calculate_tmb <- function(tumor_sample_barcode = NULL,
+                          maf_df,
+                          bed_ranges) {
+  # Calculate Tumor Mutational Burden a given sample in `Tumor_Sample_Barcode`
+  # given a target region BED frame. This function uses `snv_ranges_filter` to
+  # filter out SNVs outside those target BED ranges and uses the total size in
+  # bp of those BED ranges for the TMB denominator.
```

Comment:
My quote there was trying to ask whether I should leave out those column references from this function and tack on those columns outside this function. Is that what you are talking about @cbethell ?

Reply:
Yes, that is correct.

Reply:
I don't have a problem with this, though personally I would probably just leave it to the user to tack on histology outside the function. Mostly because who knows what the next user will want (`short_histology` vs `integrated_diagnosis` for example). But the space used by including some things is small, so I don't think it maters. But I think including `experimental_strategy` makes sense, since it is integral to interpretation of the results.